### PR TITLE
PR111 hot fix for compiler errors

### PR DESCRIPTION
**Pull request by cyanmcg**
This PR includes the quick fix of the compiler errors, which can possibly be reproduced in Fedora but not Ubuntu machines due to the different versions of compiler/linker

### CODE REVIEW

File: src/dg/strong_dg.cpp [hunk 1]
```diff
@@ -1714,11 +1714,23 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_face_term_strong(
     // the outward reference normal dircetion.
     const dealii::Tensor<1,dim,double> unit_ref_normal_int = dealii::GeometryInfo<dim>::unit_normal_vector[iface];
     // Extract the reference direction that is outward facing on the facet.
-    const int dim_not_zero = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
-                           : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 
-                           : (abs(unit_ref_normal_int[2]) >= 1e-11 ? 2 : (1000)) );
+    int dim_not_zero_probe;
+    if constexpr(dim==1){
+      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 : 1000;
+    }else if(dim==2){
```
Comment (jbrillon) [inline src/dg/strong_dg.cpp:1720]: use constexpr

File: src/dg/strong_dg.cpp [hunk 2]
```diff
@@ -1714,11 +1714,23 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_face_term_strong(
...
+      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
+                     : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 : 1000);
+    }else if(dim==3){
```
Comment (jbrillon) [inline src/dg/strong_dg.cpp:1723]: use constexpr

File: src/dg/strong_dg.cpp [hunk 3]
```diff
@@ -1714,11 +1714,23 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_face_term_strong(
...
+                         : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 
+                         : (abs(unit_ref_normal_int[2]) >= 1e-11 ? 2 : 1000));
+    }else{
+      pcout<<"Error with normals. Assume dim <=3."<<std::endl;
+      std::abort();
+    }
+    const int dim_not_zero = dim_not_zero_probe;
     if(dim_not_zero == 1000){
-        pcout<<"Error with normals. Assume dim <=3."<<std::endl;
+        pcout<<"Error with normals. Normal direction is not defined."<<std::endl;
```
Comment (jbrillon) [inline src/dg/strong_dg.cpp:1733]:
```suggestion
        pcout<<"Error with normals. Normal direction is not defined. Aborting..."<<std::endl;
```

**Review by jbrillon — APPROVED** [review]
Couple comments to resolve, thanks for the fix

File: src/dg/strong_dg.cpp (hunk 1 above)
Comment (jbrillon) [inline src/dg/strong_dg.cpp:1720]:
```suggestion
    }else if constexpr(dim==2){
```

File: src/dg/strong_dg.cpp (hunk 2 above)
Comment (jbrillon) [inline src/dg/strong_dg.cpp:1723]:
```suggestion
    }else if constexpr(dim==3){
```

**Review by jbrillon — APPROVED** [review]
addressing my comments haha

**Comment by jbrillon** [comment]
`Test #111: 2SURFACE_CONFORMING_TEST .........   Passed    0.85 sec`
`Test #112: 3SURFACE_CONFORMING_TEST .........   Passed    9.37 sec`

File: src/dg/strong_dg.cpp [hunk 4]
```diff
@@ -1714,11 +1714,23 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_face_term_strong(
...
+      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 : 1000;
+    }else if constexpr(dim==2){
+      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
+                     : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 : 1000);
+    }else if constexpr(dim==3){
+      dim_not_zero_probe = abs(unit_ref_normal_int[0]) >= 1e-11 ? 0 
+                         : (abs(unit_ref_normal_int[1]) >= 1e-11 ? 1 
+                         : (abs(unit_ref_normal_int[2]) >= 1e-11 ? 2 : 1000));
+    }else{
+      pcout<<"Error with normals. Assume dim <=3."<<std::endl;
```
Comment (cpethrick) [inline src/dg/strong_dg.cpp:1728]:
```suggestion
      pcout<<"Error with normals. Assume dim <=3. Aborting..."<<std::endl;
```

**Review by cpethrick — APPROVED** [review]
I made a small change to one of the error messages so it's more consistent. Other than that, LGTM!

**Pull request merged**